monai/apps/reconstruction/mri_dictionary.py

```diff
@@ -0,0 +1,285 @@
+
+        super().__init__(keys, cropper=None, allow_missing_keys=allow_missing_keys)  # type: ignore
+
+    def __call__(self, data: Mapping[Hashable, Tensor]) -> Dict[Hashable, Tensor]:
+        """
+        Args:
+            data: is a dictionary containing (key,value) pairs from
+                the loaded dataset
+
+        Returns:
+            the new data dictionary
+        """
+        d = dict(data)
+        for key in self.key_iterator(d):
+            # key is typically just "kspace_masked_ifft" for fastMRI data
+            image = d[key]
+            roi_size = d["target"].shape[1:]
+            roi_center = tuple(i // 2 for i in image.shape[1:])
+            cropper = SpatialCrop(roi_center=roi_center, roi_size=roi_size)
+            d[key] = convert_to_tensor(cropper(d[key]))
+        return d
+
+
+class DetailedNormalizeIntensityd(MapTransform):
+    """
+    Dictionary-based wrapper of
+    :py:class:`monai.transforms.NormalizeIntensity`.
+    This is similar to :py:class:`monai.transforms.NormalizeIntensityd`
+    but is different since (1) it automatically normalizes the target based
+    on the data, and (2) it also returns mean and std after normalization. The
+    first dimension is reserved for channel (e.g., could be num_slices).
+    The rest of the dimensions should be spatial. So the shape is (C,H,W,D)
+    for 3D samples.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: monai.transforms.MapTransform
+        subtrahend: the amount to subtract by (usually the mean)
+        divisor: the amount to divide by (usually the standard deviation)
+        nonzero: whether only normalize non-zero values.
+        channel_wise: if True, calculate on each channel separately,
+            otherwise, calculate on the entire image directly. default
+            to False.
+        dtype: output data type, if None, same as input image. defaults
+            to float32.
+        allow_missing_keys: don't raise exception if key is missing.
+    """
+
+    backend = NormalizeIntensity.backend
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        subtrahend: Optional[NdarrayOrTensor] = None,
+        divisor: Optional[NdarrayOrTensor] = None,
+        nonzero: bool = False,
+        channel_wise: bool = False,
+        dtype: DtypeLike = np.float32,
+        allow_missing_keys: bool = False,
+    ) -> None:
+        super().__init__(keys, allow_missing_keys)
+        self.default_normalizer = NormalizeIntensity(subtrahend, divisor, nonzero, channel_wise, dtype)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        """
+        Args:
+            data: is a dictionary containing (key,value) pairs from
+                the loaded dataset
+
+        Returns:
+            the new data dictionary
+        """
+        d = dict(data)
```

Comment:
I was wondering if you could add more in-line comments inside the code...Thanks!